connect_ext_test.go

```diff
@@ -2968,6 +2968,328 @@ func TestSetProtocolHeaders(t *testing.T) {
 	}
 }
 
+func TestCallInfoHeadersOnError(t *testing.T) {
+	t.Parallel()
+
+	// Handler sets up resopnse headers to test error propagation.
+	handler := &pluggablePingServerSimple{
+		ping: func(ctx context.Context, request *pingv1.PingRequest) (*pingv1.PingResponse, error) {
+			callInfo, ok := connect.CallInfoForHandlerContext(ctx)
+			if !ok {
+				return nil, connect.NewError(connect.CodeInternal, nil)
+			}
+			if request.GetNumber() < 0 {
+				callInfo.ResponseHeader().Set("x-custom-key", "ping-error")
+				callInfo.ResponseHeader().Set("x-callinfo-only", "from-callinfo")
```

Comment:
Seems like we might also want a `x-callinfo-and-error` key that is set from both and expected to be merged.

We should also be explicitly setting `callInfo.ResponseTrailer()` metadata (for both success and error cases).